lms/djangoapps/discussion_api/tests/test_api.py

```diff
@@ -448,6 +462,23 @@ def test_thread_content(self):
             }
         )
 
+    @ddt.data(
+        (FORUM_ROLE_ADMINISTRATOR, True, False),
+        (FORUM_ROLE_MODERATOR, True, False),
+        (FORUM_ROLE_COMMUNITY_TA, True, False),
+        (FORUM_ROLE_STUDENT, True, True),
+        (FORUM_ROLE_STUDENT, False, False),
+    )
```

Comment:
For comprehensive testing, would you want to add tests for `course_is_cohorted` set to `False` for each of the privileged users?  This would allow you to make sure the privileged users have the same access for cohorted courses.  Consider using `itertools.product` to test all permutations.